src/Controller/ModulesController.php

```diff
@@ -430,6 +430,61 @@ public function clone($id): ?Response
         return null;
     }
 
+    /**
+     * Clone an object from a specific point of the history.
+     *
+     * @param string|int $id Object ID.
+     * @param string|int $historyId History object ID.
+     * @return \Cake\Http\Response|null
+     */
+    public function cloneFromHistory($id, $historyId): ?Response
+    {
+        $this->viewBuilder()->setTemplate('view');
+
+        $schema = $this->Schema->getSchema();
+        if (!is_array($schema)) {
+            $this->Flash->error(__('Cannot create abstract objects or objects without schema'));
+
+            return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType]);
+        }
+        try {
+            $historyResponse = $this->apiClient->get('/history', ['filter' => ['resource_id' => $id]]);
```

Comment:
Using this query string array you don't need `usort` and `array_slice` after...

```php
  'filter' => [
     'resource_id' => $id
     'resource_type' => 'objects',
     'id' => ['ge' => $historyId],
  ],
  'sort' => 'created',
]
```

Reply:
At least a `page_size` query should also be added, history could be long...

Reply:
@stefanorosanelli for the `uname` point, is it fine to process it in an hard-coded way? It doesn't appear in the history creation entry

Reply:
@DarioSacco since `uname` has to be unique the safest thing to do is probably to unset it and force an automatic creation